core/src/term/mod.rs

```diff
@@ -1088,6 +1088,10 @@ pub enum UnaryOp {
     /// polymorphic contracts to check if they are being invoked to generate
     /// a dual contract, as part of the preliminary fix for [#1161](https://github.com/tweag/nickel/issues/1161).
     Dualize(),
+
+    /// Evaluate a string of nix code into a resulting nickel value. Currently completely (strictly) evaluates the nix code, and must result in a value serializable into JSON.
+    #[cfg(feature = "nix-experimental")]
+    EvalNix(),
```

Comment:
If we gate the existence of this variant behind `nix-experimental` but we aren't able to gate the lalrpop parser, then the code probably won't compile without `nix-experimental` anymore. I suppose the workaround would be to not gate the existence of `UnaryOp::EvalNix` but use `unimplemented!()` when evaluating in case `nix-experimental` isn't enabled.

Reply:
It now works with the feature on or off. If the feature is off it will produce an error saying that you have to recompile with `--feature nix-experimental`